### PR TITLE
feat: adds GrantRoundDonations entity sourced from GrantRounds

### PR DESCRIPTION
**Pull request by gdixon**
This PR adds the `GrantRound` as a template so that we can watch events that are emitted from the `GrantRounds` that are created via the `GrantRoundManager`. This PR also adds `GrantRoundDonations` as an entity to retrieve all contributions sent to a specific round.

--

Relates to https://github.com/dcgtc/dgrants/issues/322#issuecomment-988965316

--

A deployed example of this subgraph can be found here: https://thegraph.com/hosted-service/subgraph/gdixon/dgrants-graph-rinkeby?version=current

### CODE REVIEW

File: src/grantRound.ts
Comment (mds1) [inline src/grantRound.ts:10]: This is just me being unfamiliar with subgraphs, but what is going on here exactly / how is this working? Not sure what `dataSource` and `context` are here

File: src/grantRound.ts
Comment (mds1) [inline src/grantRound.ts:24]: Does it make sense to also include `entity.token` with the token address? It's arguably redundant but could be convenient to have it here? Will defer to you @gdixon on whether it's worth adding!

File: src/grantRound.ts
Comment (mds1) [reply src/grantRound.ts:10]: Though I probably could just read https://thegraph.academy/developers/defining-a-subgraph/ and figure it out instead of being lazy and asking you haha 😅

**Review by mds1 — APPROVED** [review]

File: src/grantRound.ts
Comment (gdixon) [reply src/grantRound.ts:10]: I think its in step 9 of that documentation! Essentially, when we create the contract listeners for the GrantRound otf, we do so with a context object, which allows us to pass over arbitrary data to the listeners, like the init args/contract address etc.

File: src/grantRound.ts
Comment (gdixon) [reply src/grantRound.ts:24]: We could do! I figured that address is already a part of the data we have available in the app, but it isn't anywhere in the Graphs data - so it does make sense! I'll see about getting it added later!

Comment (gdixon) [reply src/grantRound.ts:24]: (If we do add it, it would go unused in the App - so I don't think we should let this block merging/deploying these changes)

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**
